src/xpcc/driver/position/vl53l0_impl.hpp

```diff
@@ -0,0 +1,633 @@
+		
+		if(spadIndex < offset || spadCount == 0) {
+			calibrationSpadMap[byteIndex] &= ~(1 << bitIndex);
+		} else if (availableSpadMap[byteIndex] & (1 << bitIndex)) {
+			calibrationSpadMap[byteIndex] |= (1 << bitIndex);
+			--spadCount;
+		}
+	}
+
+	// error: less SPADs available than requested
+	if(spadCount > 0) {
+		return false;
+	}
+	
+	return true;
+}
+
+template < typename I2cMaster >
+xpcc::ResumableResult<bool>
+xpcc::Vl53l0<I2cMaster>::readSequenceInfo()
+{
+	RF_BEGIN();
+
+	// read vcsel periods
+	VL53L0_RF_CALL(read(Register::PRE_RANGE__CONFIG_VCSEL_PERIOD, sequenceInfo.vcselPeriodPreRange));
+	VL53L0_RF_CALL(read(Register::FINAL_RANGE__CONFIG_VCSEL_PERIOD, sequenceInfo.vcselPeriodFinalRange));
+
+	// decode vcsel period values
+	sequenceInfo.vcselPeriodPreRange += 1;
+	sequenceInfo.vcselPeriodPreRange <<= 1;
+	sequenceInfo.vcselPeriodFinalRange += 1;
+	sequenceInfo.vcselPeriodFinalRange <<= 1;
+
+	VL53L0_RF_CALL(read(Register::SYSTEM__SEQUENCE_CONFIG, i2cBuffer[1]));
+	sequenceInfo.enabledSteps = static_cast<MeasurementSequenceStep>(i2cBuffer[1]);
+
+	VL53L0_RF_CALL(read(Register::MSRC__CONFIG_TIMEOUT_MACROP, sequenceInfo.msrcDssTccTimeout));
+	sequenceInfo.msrcDssTccTimeout += 1;
+
+	VL53L0_RF_CALL(read(Register::PRE_RANGE__CONFIG_TIMEOUT_MACROP_HI, &i2cBuffer[1], 2));
+	sequenceInfo.preRangeTimeout = decodeTimeout((i2cBuffer[1] << 8) + i2cBuffer[2]);
+
+	VL53L0_RF_CALL(read(Register::FINAL_RANGE__CONFIG_TIMEOUT_MACROP_HI, &i2cBuffer[1], 2));	
+	sequenceInfo.finalRangeTimeout = decodeTimeout((i2cBuffer[1] << 8) + i2cBuffer[2]);
+
+	RF_END_RETURN(true);
+}
+
+template < typename I2cMaster >
+bool
+xpcc::Vl53l0<I2cMaster>::calculateFinalRangeTimeout(uint32_t measurementTime, uint16_t& registerValue)
+{
+	if(measurementTime < TimeOverhead::Start + TimeOverhead::End) {
+		return false;
+	}
+
+	// available time for "final range" step in microseconds
+	uint32_t finalRangeTime = measurementTime - TimeOverhead::Start - TimeOverhead::End;
+	const uint16_t msrcDssTccTimeout = timeoutToMicroseconds(sequenceInfo.msrcDssTccTimeout,
+															 sequenceInfo.vcselPeriodPreRange);
+
+	if(sequenceInfo.enabledSteps & MeasurementSequenceStep::TCC) {
+		const uint16_t stepTime = TimeOverhead::TCC + msrcDssTccTimeout;
+		if(finalRangeTime <= stepTime) {
+			return false;
+		}
+		finalRangeTime -= stepTime;
+	}
+
+	// "dynamic spad selection" is enabled
+	if(sequenceInfo.enabledSteps & MeasurementSequenceStep::DSS) {
+		const uint16_t stepTime = 2 * (TimeOverhead::DSS + msrcDssTccTimeout);
+		if(finalRangeTime <= stepTime) {
+			return false;
+		}
+
+		finalRangeTime -= stepTime;
+	// additional time for msrc is only needed if dss is disabled
+	} else if(sequenceInfo.enabledSteps & MeasurementSequenceStep::MSRC) {
+		const uint16_t stepTime = TimeOverhead::MSRC + msrcDssTccTimeout;
+		if(finalRangeTime <= stepTime) {
+			return false;
+		}
+
+		finalRangeTime -= stepTime;
+	}
+
+	if(sequenceInfo.enabledSteps & MeasurementSequenceStep::PreRange) {
```

Comment:
here and above, for multi-liners please put the `{` on the next line. [See source file conventions](https://github.com/roboterclubaachen/xpcc/blob/develop/doc/coding_convention.md#examples).